bliss/catalog.py

```diff
@@ -116,7 +116,13 @@ class TileCatalog(BaseTileCatalog):
     galaxy_params_index = {k: i for i, k in enumerate(galaxy_params)}
 
     def __init__(self, tile_slen: int, d: Dict[str, Tensor]):
-        self.max_sources = d["locs"].shape[3]
+        self.locations_sources = True
+
+        if "locs" in d:
+            self.max_sources = d.get("locs").shape[3]
```

Comment:
Could you use `BaseTileCatalog` instead of adding special cases to `TileCatalog`?

Reply:
will take a look. The main issue here and in the previous case is that TileCatalog assumes ''locs'' is in the data, but we provide no such parameter

Reply:
resolved in refactor